kernel/src/main/java/org/kframework/compile/checks/CheckHOLE.java

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019 K Team. All Rights Reserved.
+package org.kframework.compile.checks;
+
+import org.kframework.builtin.Sorts;
+import org.kframework.compile.ResolveStrict;
+import org.kframework.definition.Context;
+import org.kframework.definition.Module;
+import org.kframework.definition.Production;
+import org.kframework.definition.Sentence;
+import org.kframework.kore.K;
+import org.kframework.kore.KApply;
+import org.kframework.kore.KVariable;
+import org.kframework.kore.VisitK;
+import org.kframework.utils.errorsystem.KEMException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static org.kframework.kore.KORE.*;
+
+public class CheckHOLE {
+    private final Set<KEMException> errors;
+    private final Module m;
+
+    public CheckHOLE(Set<KEMException> errors, Module m) {
+        this.errors = errors;
+        this.m = m;
+    }
+
+    public void check(Sentence sentence) {
+        if (sentence instanceof Production) {
+            check((Production) sentence);
+        } else if (sentence instanceof Context) {
+            check((Context) sentence);
+        }
+    }
+
+    private void check(Production prod) {
+        if (prod.att().contains("strict")) {
+            check(prod, prod.att().get("strict"));
+        }
+        if (prod.att().contains("seqstrict")) {
+            check(prod, prod.att().get("seqstrict"));
+        }
+    }
+
+    private void check(Production prod, String att) {
+        long arity = prod.nonterminals().size();
+        List<Integer> strictnessPositions = new ArrayList<>();
+        if (att.isEmpty()) {
+            for (int i = 1; i <= arity; i++) {
+                strictnessPositions.add(i);
+            }
+        } else {
+            try {
+                String[] components = att.split(";");
+                if (components.length == 1) {
+                    if (Character.isDigit(components[0].trim().charAt(0))) {
+                        ResolveStrict.setPositions(components[0].trim(), strictnessPositions, arity, prod);
+                    } else {
+                        for (int i = 1; i <= arity; i++) {
+                            strictnessPositions.add(i);
+                        }
+                    }
+                } else if (components.length == 2) {
+                    ResolveStrict.setPositions(components[1].trim(), strictnessPositions, arity, prod);
+                } else {
+                    errors.add(KEMException.compilerError("Invalid strict attribute containing multiple semicolons.", prod));
```

Comment:
This is already checked in `ResolveStrict.java` above. Is it possible to only have it checked once?